libs/layout/src/lib/abstracts/drag-and-drop/drag-and-drop.service.ts

```diff
@@ -0,0 +1,157 @@
+
+	/**
+	 * A custom set of messages used for the drag and drop events.
+	 *
+	 *  Please check the readme for more information on what is necessary to make these messages accessible.
+	 */
+	public customMessages: Record<string, NgxAccessibleDragAndDropMessageRecord>;
+
+	/**
+	 * Sets a message to the live region
+	 *
+	 * @param message - The provided message
+	 */
+	public setMessage(message: NgxAccessibleDragAndDropMessage): Observable<void> {
+		// Iben: Take the current language to fetch the message
+		return (
+			typeof this.currentLanguage === 'string'
+				? of(this.currentLanguage)
+				: this.currentLanguage
+		).pipe(
+			filter(Boolean),
+			take(1),
+			tap((currentLanguage) => {
+				// Iben: Fetch the necessary data
+				const { type, data } = message;
+
+				let result: string = this.messageRecord[currentLanguage][type];
+
+				// Iben: If no message was found, we early exit and throw an error
+				if (!result) {
+					console.error(
+						'NgxAccessibleDragAndDropAbstractService: No message for the corresponding drag and drop event was found.'
+					);
+
+					return;
+				}
+
+				// Iben: Replace the necessary substrings
+				if (type === 'selected' || type === 'deselected' || type === 'cancelled') {
+					result = result.replace(
+						'{{#item}}',
+						data.itemLabel || `${this.messageRecord[currentLanguage].item} ${data.item}`
+					);
+				} else if (type === 'moved') {
+					console.dir(result);
+					result = result
+						.replace(
+							'{{#item}}',
+							data.itemLabel ||
+								`${this.messageRecord[currentLanguage].item} ${data.item}`
+						)
+						.replace(
+							`{{#from}}`,
+							data.fromLabel ||
+								`${this.messageRecord[currentLanguage].container} ${data.from}`
+						)
+						.replace(
+							`{{#to}}`,
+							data.toLabel ||
+								`${this.messageRecord[currentLanguage].container} ${data.to}`
+						);
+				} else if (type === 'reordered') {
+					result = result
+						.replace(
+							'{{#item}}',
+							data.itemLabel ||
+								`${this.messageRecord[currentLanguage].item} ${data.item}`
+						)
+						.replace(`{{#from}}`, data.from)
+						.replace(`{{#to}}`, data.to);
+				}
+
+				// Iben: Update the message in the live region
+				this.liveRegionService.setMessage(result);
+			}),
+			map(() => null)
+		);
+	}
+
+	/**
+	 * Adds a description to the drag and drop host explaining how the drag and drop functions
+	 *
+	 * @param  parent - The drag and drop host
+	 * @param description - An optional description used to overwrite the default description
+	 */
+	public setDragAndDropDescription(parent: HTMLElement, description?: string): Observable<void> {
+		// Iben: Create the description element and its id
+		const element: HTMLParagraphElement = document.createElement('p');
+		const id: string = UUID.UUID();
+
+		// Iben: Take the current language to fetch the message
+		return (
+			typeof this.currentLanguage === 'string'
```

Comment:
What if this evaluates as a non-string value, but the value is not an observable (for example `undefined` or something else)?

Reply:
I get what you mean here, and I agree on checking for null/undefined; but the typing will already prevent you from returning anything but a string or an Observable. If you decide to bypass this by casting to any then that's on you 😂 

`abstract get currentLanguage(): string | Observable<string>;`